apps/crn-server/test/handlers/external-author/index-handler.test.ts

```diff
@@ -1,28 +1,108 @@
+import { NotFoundError } from '@asap-hub/errors';
 import { indexExternalAuthorHandler } from '../../../src/handlers/external-author/index-handler';
 import {
   getExternalAuthorEvent,
   getExternalAuthorResponse,
 } from '../../fixtures/external-authors.fixtures';
 import { algoliaSearchClientMock } from '../../mocks/algolia-client.mock';
-import { externalAuthorDataProviderMock } from '../../mocks/external-author-data-provider.mock';
+import { externalAuthorControllerMock } from '../../mocks/external-author-controller.mock';
 
 describe('External Author index handler', () => {
   const indexHandler = indexExternalAuthorHandler(
-    externalAuthorDataProviderMock,
+    externalAuthorControllerMock,
     algoliaSearchClientMock,
   );
 
+  const OLD_ENV = process.env;
+
+  beforeEach(() => {
+    jest.resetModules();
+    jest.resetAllMocks();
+    process.env = { ...OLD_ENV };
+  });
+
   afterEach(() => jest.clearAllMocks());
 
+  afterAll(() => {
+    process.env = OLD_ENV;
+  });
+
+  test('Should instantiate contentful data provider when feature flag is truthy', async () => {
```

Comment:
see my comment above, i think this logic isn't correct, I think where this is indexed depends on the source. And I think this part should be blocked by my work. I suggest that you leave the live sync of Algolia out of scope for this one and just make sure that when we run on-demand algolia sync it gets indexed in the `-contentful` index. I can probably rebase on top of that and add the sync to external authors